src/core/snapshot.js

```diff
@@ -9,22 +9,44 @@ const typeOf = { value: Symbol.for('react.test.json') }
 // Clean up result for better looking on CodeMirror
 const cleanUp = (result) => {
   result.actual = result.actual.trim()
-  result.expected = result.expected.trim()
+  result.expected = (result.expected || '').trim()
   result.diff = (result.diff || '').replace(/\n[ ]+\n/g, '\n\n')
 }
 
-const snapshot = (name, tree, update) => {
-  const destination = path.resolve(base, `${name}.snap`)
-  const state = new SnapshotState(null, update, destination)
+const setTreeTypes = (obj) => {
+  if (!obj) return
 
-  if (tree) {
-    Object.defineProperty(tree, '$$typeof', typeOf)
+  if (Array.isArray(obj)) {
+    obj.forEach(setTreeTypes)
+  } else if (obj.type) {
+    setTreeTypes(obj.children)
+    Object.defineProperty(obj, '$$typeof', typeOf)
   }
+}
+
+const snapshot = (name, tree, update) => {
+  const snapshotPath = path.resolve(base, `${name}.snap`)
+  const state = new SnapshotState(null, {
+    snapshotPath,
+    updateSnapshot: update ? 'all' : 'new',
+  })
+
+  setTreeTypes(tree)
 
   const result = state.match(name, tree)
-  state.save(update)
 
-  if (!result.pass) {
+  // NOTE: result.expected can be null
+  result.expected = String(result.expected)
+
+  state.save()
+
+  // Keep backward compatible; remove extra props added to passing result
+  // in newest jest-snapshot pkg
+  if (result.pass) {
```

Comment:
I don't think we need this.
If it's just to make the old tests pass (without adding the new properties in the expectations), you can use [objectContaining](https://facebook.github.io/jest/docs/en/expect.html#expectobjectcontainingobject):

```js
expect(result).toEqual(expect.objectContaining({ pass: true }))
```

Reply:
Sounds good. I wasn't sure how important it was to keep the output matching the previous release